### PR TITLE
Fix for issue #5 (make battery plugin work with /sys)

### PR DESCRIPTION
**Pull request by bubak4**
Fix for parsing battery / ac power information from /sys filesystems.  Makes battery plugin work on newer kernels.

Original patch was submitted by me (martinslouf@sourceforge.net) to sourceforge.  This is just a simple migration of the patch once fbpanel moved to github.

There are 2 commits -- (1) the enhancement itself + (2) simple test with valgrind to check if it works

### CODE REVIEW

File: plugins/battery/power_supply.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
+// Time-stamp: < power_supply.h (2015-12-04 09:42) >
+// author: bubak4@github (aka. martinslouf@users.sourceforge.net)
+
+#ifndef POWER_SUPPLY_H
+#define POWER_SUPPLY_H
+
+#include <glib-2.0/glib.h>
+
+#ifdef __cplusplus
```
Comment (aanatoly) [inline plugins/battery/power_supply.h]: plz remove "c++" ifdefs and indent all below code to column 0

File: plugins/battery/power_supply.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    /* Struct representing all the power supplies on current system. */
+    typedef struct {
+        /* List of ac structs. */
+        GSequence* ac_list;
+        /* List of bat structs. */
+        GSequence* bat_list;
+    } power_supply;
+
+    /* Allocate memory for struct power_supply. */
+    extern power_supply* power_supply_new();
```
Comment (aanatoly) [inline plugins/battery/power_supply.h]: plz remove `exten` keyword for functions - they are redundant in C-only project

**Comment by aanatoly** [comment]
Hi Martin
I follow [git-flow branching model](http://jeffkreeftmeijer.com/2010/why-arent-you-using-git-flow/) so  I'd like to merge your PR into `develop` branch rather then `master`. I can't find a way to change destination branch in Gihub web interface, so we left with 2 options:

 * I merge it from command line (I have no problem with this :-) )
 * You make PR based on `develop` branch and I merge it from web 

What do you prefer?

**Comment by bubak4** [comment]
According to project guidelines I have created a new pull request to develop branch. See #8.

**Pull request closed without merging**